src/pages/GamePage.js

```diff
@@ -142,6 +139,29 @@ function GamePage({ match }) {
     });
   }
 
+  async function callFinishGame() {
+    let timeDataRef = await finishGame({ gameId: gameId });
+    let endTime = timeDataRef.data.endedAt;
+    if (game.status === "ingame") {
+      firebase
+        .database()
+        .ref(`games/${gameId}`)
+        .update(
+          {
+            status: "done",
```

Comment:
The point of the serverless function was so it can update the game-ended time from a trusted source. As you have it currently set up, users can essentially set the endedAt time to an arbitrary value in the past, which is insecure.

You should have this database update inside the serverless function itself.

Reply:
Should be more secure now